app/browser/api/ledger.js

```diff
@@ -727,70 +687,61 @@ const updatePublishers = (state, publisherKeys) => {
 }
 
-// TODO rename function
-const excludeP = (publisherKey, callback) => {
-  let doneP
-
-  const done = (err, result) => {
-    doneP = true
-    callback(err, result)
+const shouldExclude = (publisherKey) => {
+  // Check for the .gov suffix, we should exclude governmental sites
+  const suffix = tldjs.getPublicSuffix(publisherKey)
+  if (suffix === 'gov') {
+    return true
   }
 
-  if (!v2RulesetDB) {
-    return setTimeout(() => excludeP(publisherKey, callback), 5 * ledgerUtil.milliseconds.second)
+  const domain = tldjs.getDomain(publisherKey)
+  // Checks for government site edge cases, such as aces.gov.in, gov.ie, nenkin.go.jp
+  // These were previously used in the bat-client default ruleset
+  const sldGovRegex = /^[a-z][a-z].gov$/
+  const sfxGovRegex1 = /^go.[a-z][a-z]$/
+  const sfxGovRegex2 = /^gov.[a-z][a-z]$/
+
+  if (
+    sldGovRegex.test(domain) ||
+    sfxGovRegex1.test(suffix) ||
+    sfxGovRegex2.test(suffix)
+  ) {
+    return true
   }
 
-  inspectP(v2RulesetDB, v2RulesetPath, publisherKey, 'exclude', 'domain:' + publisherKey, (err, result) => {
-    if (!err) {
-      return done(err, result)
-    }
+  // Public keybase users should also be excluded
+  if (domain === 'keybase.pub') {
+    return true
+  }
 
-    let props = ledgerPublisher.getPublisherProps(publisherKey)
-    if (!props) return done()
+  return false
+}
 
-    v2RulesetDB.createReadStream({lt: 'domain:'}).on('data', (data) => {
-      if (doneP) return
+const doneP = (err, result, callback) => {
+  callback(err, result)
+}
 
-      const sldP = data.key.indexOf('SLD:') === 0
-      const tldP = data.key.indexOf('TLD:') === 0
-      if (!tldP && !sldP) return
+const getPublisherExclude = (publisherKey, callback) => {
+  if (!publisherInfoData) {
+    return setTimeout(() => module.exports.getPublisherExclude(publisherKey, callback), 15 * ledgerUtil.milliseconds.second)
+  }
 
-      if (underscore.intersection(data.key.split(''),
-          ['^', '$', '*', '+', '?', '[', '(', '{', '|']).length === 0) {
-        if (data.key !== ('TLD:' + props.TLD) && (props.SLD && data.key !== ('SLD:' + props.SLD.split('.')[0]))) {
-          return
-        }
-      } else {
-        try {
-          const regexp = new RegExp(data.key.substr(4))
-          if (!regexp.test(props[tldP ? 'TLD' : 'SLD'])) return
-        } catch (ex) {
-          console.error(v2RulesetPath + ' stream invalid regexp ' + data.key + ': ' + ex.toString())
-        }
-      }
+  const publisher = publisherInfoData.find(p => p.first() === publisherKey)
 
-      let result
-      try {
-        result = JSON.parse(data.value)
-      } catch (ex) {
-        console.error(v2RulesetPath + ' stream invalid JSON ' + data.entry + ': ' + data.value)
-      }
+  if (typeof publisher === 'undefined') {
+    // If publisher doesn't exist in infoData, check the exclusion rules
+    const exclude = module.exports.shouldExclude(publisherKey)
+    return module.exports.doneP(null, exclude, callback)
```

Comment:
nit: can we just rename `exclude` in `result` and have one return?

Reply:
Can do

Reply:
done